### PR TITLE
[17.0][ADD] sale_order_retrieve_customer_after_delivery

### PR DESCRIPTION
**Pull request by IriaAlonso**
- Move delivery address field to before customer field
- Delivery address field retrieve customer

### CODE REVIEW

**Comment by dalonsod** [comment]
See my last commit:
* Passing context values across a compute field flow finally didn't work, so it wasn't an option.
* Indeed, adding an `onchange` method to a field that actually has a compute stored behavior (`partner_shipping_id`) didn't work. That's why partner shouldn't be filled when editing partner shipping address!
* So I've made a workaround with this approach: making a new instrumental `partner_shipping_2_id` field, the visible one, with almost the same definition than original one, that makes the magic of autofilling partner. And a last step: overwrite compute original partner shipping method, making new and old field identical (former `partner_shipping_id` is stored in background, because we need it for other processes!).

Please check that it's working, and:
* Add a pre-init hook that create basic `partner_shipping_2_id` field and initialize it with `partner_shipping_id` content. If you don't find a similiar example, please ask.
* Make translations and `README` stuff.

**Comment by IriaAlonso** [comment]
@dalonsod

File: sale_order_retrieve_customer_after_delivery/__init__.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
+from . import models
+from odoo.tools import column_exists, create_column
+
+def pre_init_hook(env):
+   
+    if not column_exists(env.cr, "sale_order", "partner_shipping_2_id"):
+        create_column(env.cr, "sale_order", "partner_shipping_2_id", "character varying")
```
Comment (dalonsod) [inline sale_order_retrieve_customer_after_delivery/__init__.py]: This is wrong, `partner_shipping_2_id` should be an `integer` value, as a `Many2one` field:
```suggestion
        create_column(env.cr, "sale_order", "partner_shipping_2_id", "integer")
```

File: sale_order_retrieve_customer_after_delivery/__init__.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
+from . import models
+from odoo.tools import column_exists, create_column
+
+def pre_init_hook(env):
```
Comment (dalonsod) [inline sale_order_retrieve_customer_after_delivery/__init__.py]: This function should be better placed in a `hooks.py` file at this level and then import it in `__init__.py` with 

`from .hooks import pre_init_hook`

See e.g. https://github.com/OCA/product-attribute/tree/17.0/product_logistics_uom

File: sale_order_retrieve_customer_after_delivery/static/description/index.html [hunk 3]
```diff
@@ -0,0 +1,417 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (dalonsod) [inline sale_order_retrieve_customer_after_delivery/static/description/index.html]: Remove this as usual in v17

**Review by dalonsod — CHANGES_REQUESTED** [review]
See comments, and also:
* `i18n` folder is wrongly placed, move it to addon root. If not, translations should never be applied.
* Add a `ROADMAP.rst` file indicating that is pending compatibility with non-delivery address mode (something stupid, installing this addon and disabling delivery address mode, but possible). Then, re-create `README.rst`.

We'll keep name's addon at this point, it's not a very good mane IMO but not terrible anyway 😉

cc @ChristianSantamaria

**Comment by IriaAlonso** [comment]
@dalonsod

File: sale_order_retrieve_customer_after_delivery/i18n/es.po [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+#. module: sale_order_retrieve_customer_after_delivery
+#: model_terms:ir.ui.view,arch_db:sale_order_retrieve_customer_after_delivery.sale_order_view_form
+msgid "Delivery Address"
+msgstr "Dirección de entrega"
+
+#. module: sale_order_retrieve_customer_after_delivery
+#: model:ir.model.fields,field_description:sale_order_retrieve_customer_after_delivery.field_sale_order__partner_shipping_2_id
+msgid "Delivery Address (new)"
+msgstr "Dirección de entrega"
```
Comment (dalonsod) [inline sale_order_retrieve_customer_after_delivery/i18n/es.po]: Better this (this label is only for field definition, so we've interested in marking it different than original one)

```suggestion
msgstr "Dirección de entrega (nueva)"
```

**Review by dalonsod — CHANGES_REQUESTED** [review]
See very minor comment, and I'll test It ASAP

**Comment by IriaAlonso** [comment]
@dalonsod done

**Comment by dalonsod** [comment]
Squash made with latest change:

imagen

This typo in view was due to a `replace` instead of `attributes` in `xpath` for `form` view, when hiding original delivery address field. Fixed.

**Comment by dalonsod** [comment]
@ChristianSantamaria could you install it at v17 CmhProd environment and make it also available for CmhTest environment?

**Pull request merged**